src/GraphQL/Internal/API/Enum.hs

```diff
@@ -86,24 +86,16 @@ instance forall conName p b. (KnownSymbol conName) => GenericEnumValues (C1 ('Me
 
 -- TODO(tom): better type errors using `n`. Also type errors for other
 -- invalid constructors.
-instance forall conName p b sa sb.
-  ( TypeError ('Text "Constructor not unary: " ':<>: 'Text conName)
+instance forall conName f s sa sb.
+  ( TypeError ('Text "Constructor not nullary: " ':<>: 'Text conName)
   , KnownSymbol conName
-  ) => GenericEnumValues (C1 ('MetaCons conName p b) (S1 sa sb)) where
-  genericEnumValues = nonUnaryConstructorError
-  genericEnumFromValue = nonUnaryConstructorError
-  genericEnumToValue = nonUnaryConstructorError
-
-instance forall conName p b sa sb f.
-  ( TypeError ('Text "Constructor not unary: " ':<>: 'Text conName)
-  , KnownSymbol conName
-  ) => GenericEnumValues (C1 ('MetaCons conName p b) (S1 sa sb) :+: f) where
+  ) => GenericEnumValues (C1 ('MetaCons conName f s) (S1 sa sb)) where
   genericEnumValues = nonUnaryConstructorError
   genericEnumFromValue = nonUnaryConstructorError
   genericEnumToValue = nonUnaryConstructorError
 
 nonUnaryConstructorError :: a
-nonUnaryConstructorError = panic "Tried to construct enum with non-unary constructor. Should get a compile-time error instead of this."
```

Comment:
Please also rename this function to `nonNullaryConstructorError`. We don't have to worry about backwards compatibility as it is not exported and in `Internal` anyway.